1_fetch/src/get_feature_vars.R

```diff
@@ -489,3 +559,246 @@ prep_feature_vars <- function(sb_var_data, sites_all, sites_screened,
+      select(-ends_with(".y"), -ID, -starts_with("..."), -contains("_S1"))
+    
+  } else {
+    
+    data <- data %>%
+      left_join(data_temp, by = "COMID")
+    
+  }
+  
+  #rename duplicated headers (if any) and replace headers containing "/" with "-"
+  rename_dup_headers <- list()
+  for (i in 1:ncol(data)) {
+    header <- names(data)[i]
+    if (str_sub(header, -2) == ".x") {
+      new_header <- str_sub(header, 1, -3)
+    } else if (str_detect(header, "/")) {
+      new_header <- str_replace(header, "/", "-")
+    } else {
+      new_header <- header
+    }
+    rename_dup_headers[[i]] <- new_header
+  }
+  rename_dup_headers <- as.character(rename_dup_headers)
+  names(data) <- rename_dup_headers
+  
+  file_out <- paste0(outdir, "/sb_", id, "_", out_file_label, ".csv")
+  write_csv(data, file_out)
+  return(file_out)
+}
+
+
+prep2_vars_conus <- function(sohl_early, sohl_late, outdir, out_file_label) {
+  
+  #'@description joins all data downloaded from ScienceBase with sites of interest
+  #'
+  #'@param prep1_vars target generated from the 'prep1_vars_conus()' function
+  #'@param outdir output directory
+  #'@param out_file_label naming convention to label output files from this function
+  #'
+  #'@return data frame with COMID column appended by all feature variables of interest
+  
+  #read in sb data
+  early <- read_csv(paste0(outdir, "/sb_", sohl_early, "_raw_conus.csv"), 
+                    show_col_types = FALSE)
+  late <- read_csv(paste0(outdir, "/sb_", sohl_late, "_raw_conus.csv"), 
+                   show_col_types = FALSE)
+  late <- late %>%
+    select(COMID, contains("SOHL00"))
+  sohl <- left_join(early, late, by = "COMID")
+  sohl[sohl == -9999] <- NA_real_
+  
+  unit <- c("CAT", "ACC", "TOT")
+  category <- as.character(1:17)
+  
+  data_all <- tibble(COMID = sohl$COMID)
+  comid_out_conus_all <- tibble()
+  for (i in unit) {
+    data_by_unit <- sohl %>%
+      select(COMID, contains(i))
+    data_all_cats <- tibble(COMID = sohl$COMID)
+    for (j in category) {
+      data_by_category <- data_by_unit %>%
+        select(COMID, ends_with(paste0("_", j))) %>%
+        mutate(longterm_avg = rowMeans(select(., -COMID))) %>%
+        select(COMID, longterm_avg)
+      name <- paste0("SOHL", j)
+      names(data_by_category) <- c("COMID", name)
+      data_all_cats <- left_join(data_all_cats, data_by_category, by = "COMID")
+    }
+    data_reclass <- data_all_cats %>%
+      mutate(WATER = SOHL1, 
+             DEVELOPED = SOHL2, 
+             FOREST = rowSums(select(., SOHL3, SOHL4, SOHL5, SOHL8, SOHL9, SOHL10), 
+                                na.rm = TRUE),
+             MINING = SOHL6, 
+             BARREN = SOHL7, 
+             GRASSLAND = SOHL11, 
+             SHRUBLAND = SOHL12, 
+             CROPLAND = SOHL13, 
+             HAYPASTURE = SOHL14, 
+             WETLAND = rowSums(select(., SOHL15, SOHL16),
+                                 na.rm = TRUE), 
+             ICESNOW = SOHL17) %>%
```

Comment:
In this function, do `HAYPASTURE` and `ICESNOW` need to be edited here to have `-`? Does the g2 processing function also need this edit?